include/dlaf/common/index2d.h

```diff
@@ -139,25 +143,31 @@ class Index2D : public internal::basic_coords<IndexT> {
   using IndexType = IndexT;
 
   /// Create an invalid 2D coordinate
-  Index2D() noexcept;
+  Index2D() noexcept : internal::basic_coords<IndexT>(-1, -1) {}
 
   /// Create a valid 2D coordinate
   /// @param row index of the row
   /// @param col index of the column
   /// @throw std::invalid_argument if row < 0 or col < 0
-  Index2D(IndexT row, IndexT col);
+  Index2D(IndexT row, IndexT col) : internal::basic_coords<IndexT>(row, col) {
+    if (!internal::basic_coords<IndexT>::isValid())
+      throw std::invalid_argument("indices are not valid (negative).");
```

Comment:
Can we make this an ASSERT_* ?

Reply:
I didn't change it because it is not the main topic of this PR, and the replacement with the new tools is going on in #178